src/java.desktop/share/classes/java/awt/color/ICC_Profile.java

```diff
@@ -1085,8 +1099,8 @@ public void write(OutputStream s) throws IOException {
      * @see #setData(int, byte[])
      */
     public byte[] getData(int tagSignature) {
-        activate();
-        return getData(cmmProfile, tagSignature);
+        byte[] t = getData(cmmProfile(), tagSignature);
+        return t != null ? t.clone() : null;
```

Comment:
I have moved the clone operation to the public "ICC_Profile.getData(int tagSignature)" method, so we do not clone it again again and again when we use this data internally.